File: old/examples/1BillionBenchmark/billion.rb
def epoch
  Time.now.to_f
end

start = epoch

total = 0
(1..1000000000).each do |i|
  total += i * i
end

taken = epoch - start

puts "Ruby took #{'%.9f' % taken} seconds to calculate the sum of squares from 1 to 1 billion:"
puts "Total: #{total}"

File: old/examples/1BillionBenchmark/billion.ts
function epoch(): number {
    return Date.now() / 1000;
}

async function main() {
    const started = epoch();

    let total = BigInt(0);
    for (let i = BigInt(1); i <= BigInt(1000000000); i++) {
        total += BigInt(i * i);
    }

    const taken = epoch() - started;

    console.log(`TypeScript took ${taken.toFixed(9)} seconds to calculate the sum of squares from 1 to 1 billion:`);
    console.log(`Total: ${total.toString()}`);
}

main();